File: tests/go/cases/case_150_closure_no_escape_05.go
package escape_tests

var retainedCase150 = []map[string]string{}

func Case150ClosureNoEscape05(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "closure_no_escape_05",
		"entity": "stress",
		"stage": "evaluation",
		"input": raw,
		"checksum": "closure_no_escape_05:" + raw,
	}
	consume := func() string { return payload["task"] }
	_ = consume()
	// SAFE: closure invoked locally only.
	return payload["checksum"]
}

File: tests/nodejs/cases/case_150_closure_no_escape_05.js
/**
 * Task case 150: closure_no_escape_05 false-positive/false-negative stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case150ClosureNoEscape05(input) {
  const taskName = 'closure_no_escape_05';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'stress',
    stage: 'evaluation',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  const consume = () => payload.task;
  void consume();
  // SAFE: closure invoked locally and not retained.
  return payload.checksum;
}

module.exports = {
  case150ClosureNoEscape05,
};
